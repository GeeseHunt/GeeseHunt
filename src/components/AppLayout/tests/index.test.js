/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { AppLayout } from '..';

describe('AppLayout', () => {
  let shallow;
  let props;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    const fakeUser = {
      id: 'id',
      displayName: 'name',
      avatarUrl: '',
    };
    props = { user: fakeUser };
  });

  describe('#render', () => {
    it('renders without errors', () => {
      const wrapper = shallow(
        <AppLayout {...props}>
          <div>Hello world!</div>
        </AppLayout>,
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
