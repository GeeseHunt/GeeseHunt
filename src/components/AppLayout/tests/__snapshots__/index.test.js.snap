// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout #render renders without errors 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(CssBaseline) />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-2"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="open drawer"
        className="makeStyles-menuButton-4"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-appTitle-12"
        noWrap={true}
        variant="h6"
      >
        GeeseHunt
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(IconButton))
        size="small"
      >
        <WithStyles(ForwardRef(Avatar))
          alt="name"
          className="makeStyles-avatar-14"
          src=""
        />
      </WithStyles(ForwardRef(IconButton))>
      <LoginDialog
        onClose={[Function]}
        open={false}
      />
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    className="makeStyles-drawer-6 makeStyles-drawerClose-8"
    classes={
      Object {
        "paper": "makeStyles-drawerClose-8",
      }
    }
    variant="permanent"
  >
    <div
      className="makeStyles-toolbar-9"
    >
      <WithStyles(ForwardRef(IconButton))
        onClick={[Function]}
      >
        <ChevronLeftIcon />
      </WithStyles(ForwardRef(IconButton))>
    </div>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(List))>
      <Link
        className="makeStyles-listItemLink-13"
        key="Explorer Course"
        onClick={null}
        to="/explore"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <ExploreIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Explorer Course"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
      <Link
        className="makeStyles-listItemLink-13"
        key="My GPA"
        onClick={null}
        to="/"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <SpellcheckIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="My GPA"
          />
        </WithStyles(ForwardRef(ListItem))>
      </Link>
    </WithStyles(ForwardRef(List))>
  </WithStyles(ForwardRef(Drawer))>
  <main
    className="makeStyles-content-10"
  >
    <div
      className="makeStyles-toolbar-9"
    />
    <div
      className="makeStyles-children-11"
    >
      <div>
        Hello world!
      </div>
    </div>
  </main>
</div>
`;
